README.md

```diff
@@ -3,301 +3,284 @@
-  delay: 0,               // Delay milliseconds when shooting screenshots
-  waitFor: '',            // User defined trigger function name to shoot screenshots. See "Full control the screenshot timing" section below.
-  viewport: {             // Browser's viewport when shooting screenshots. (See: https://github.com/GoogleChrome/puppeteer/blob/master/docs/api.md#pagesetviewportviewport)
-    width: 1024,
-    height: 768,
-    deviceScaleFactor: 1,
-    isMobile: false,
-    hasTouch: false,
-    isLandscape: false,
-  },
-  filePattern: '{kind}-{story}-{knobs}_{ns}-{vp}' // file pattern, combined from kind, story name, used knobs, namespace and viewport
+`ScreenShotOptions` object is available as the value of the key `screenshot` of `addParameters` argument or `withScreenshot` argument.
+
+```ts
+interface ScreenshotOptions {
+  delay?: number;                           // default 0 msec
+  waitImages?: boolean;                     // default true
+  waitFor?: string | () => Promise<void>;   // default ""
+  fullPage?: boolean;                       // default true
+  hover?: string;                           // default ""
+  focus?: string;                           // default ""
+  skip?: boolean;                           // default false
+  viewport?: Viewport;
+  viewports?: string[] | { [variantName]: Viewport };
+  variants?: Variants;
 }
 ```
 
-Also, By passing the `array` to `viewport`, you can easily shoot multiple Viewports.
-
-```javascript
-{
-  viewport: [
-    // Mobile
-    {
-      width: 300,
-      height: 420,
-      isMobile: true,
-      hasTouch: true,
-    },
-    // Tablet
-    {
-      width: 768,
-      height: 800,
-      isMobile: true,
-      hasTouch: true,
-    },
-    // Desktop
-    {
-      width: 1024,
-      height: 768,
-    },
-  ],
-}
+- `delay`: Waiting time [msec] before capturing.
+- `waitImages`: If set true, Storycap waits until `<img>` in the story are loaded.
+- `waitFor` : If you set a function to return `Promise`, Storycap waits the promise is resolved. You can also set a name of global function that returns `Promise`.
+- `fullPage`: If set true, Storycap captures the entire page of stories.
+- `focus`: If set a valid CSS selector string, Storycap captures after focusing the element matched by the selector.
+- `hover`: If set a valid CSS selector string, Storycap captures after hovering the element matched by the selector.
+- `skip`: If set true, Storycap cancels capturing corresponding stories.
+- `viewport`, `viewports`: See type `Viewport` section below.
+- `variants`: See type `Variants` section below.
+
+### type `Variants`
+
+`Variants` is used to generate [multiple PNGs from 1 story](#multiple-pngs-from-1-story).
+
+```ts
+type Variants = {
+  [variantName: string]: {
+    extends?: string | string[]; // default: ""
+    delay?: number;
+    waitImages?: boolean;
+    waitFor?: string | () => Promise<void>;
+    fullPage?: boolean;
+    hover?: string;
+    focus?: string;
```

Comment:
I would like to add the following functions. I think we can deal with the uniformity of the API and wider user needs :dog:

- `:visited`
- `:active`

Reply:
Absolutely I agree. However, Puppeteer does not expose function to change an element's status to `:visited` nor `:active` . And I don't  know how to do this using `elementHandler` and DOM API 😭 ... Do you have any idea to implement this ?

Reply:
It seems hard to deal with, I don't have any idea of implementation... :innocent:
(Chrome Developer Tools has functions that can be applied pseudo-classes, It's a pity that it seems to be possible...)